Splitting pane with extensions tab open crashes Zed

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Attempting to open a split pane from the command palette when the Extensions tab is open results in a crash.

This happens with and without any extensions installed.

### Environment

Zed: v0.123.1 (Zed Preview)
OS: macOS 14.2.1
Memory: 16 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature


https://github.com/zed-industries/zed/assets/16290753/6938d7ab-9186-4cda-b661-cefbf7163d73



### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.


<details>
  <summary>Expand Zed logs</summary>
  2024-02-15T10:16:30-05:00 [INFO] ========== starting zed ==========
  2024-02-15T10:16:30-05:00 [INFO] Opening main db
  2024-02-15T10:16:30-05:00 [ERROR] crates/theme/src/settings.rs:344: theme not found: Vesper
  2024-02-15T10:16:30-05:00 [ERROR] crates/theme/src/settings.rs:212: theme not found: Vesper
  2024-02-15T10:16:30-05:00 [INFO] Opening main db
  2024-02-15T10:16:30-05:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:270: expected value at line 1 column 1
  2024-02-15T10:16:30-05:00 [INFO] build git repository ".git"
  2024-02-15T10:16:30-05:00 [INFO] set status on client 0: Authenticating
  2024-02-15T10:16:31-05:00 [INFO] set environment variables from shell:/bin/zsh, path:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Library/Apple/usr/bin:/<path>/.fig/bin:/<path>/.local/bin:/<path>/.orbstack/bin:/<path>/.local/share/mise/shims:/<path>/.modular/pkg/packages.modular.com_mojo/bin
  2024-02-15T10:16:31-05:00 [INFO] set status on client 57425: Connecting
  2024-02-15T10:16:31-05:00 [INFO] Opening main db
  2024-02-15T10:16:31-05:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
  2024-02-15T10:16:31-05:00 [INFO] Node runtime install_if_needed
  2024-02-15T10:16:31-05:00 [INFO] add connection to peer
  2024-02-15T10:16:31-05:00 [INFO] waiting for server hello
  2024-02-15T10:16:31-05:00 [INFO] got server hello
  2024-02-15T10:16:31-05:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 332, id: 609161 })
  2024-02-15T10:16:31-05:00 [INFO] set status on client 57425: Connected { peer_id: PeerId { owner_id: 332, id: 609161 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
  2024-02-15T10:16:32-05:00 [INFO] 0 unhandled notification LogMessage:
  {
    "level": 0,
    "message": "[DEBUG] [agent] [2024-02-15T15:16:32.355Z] Agent service starting",
    "metadataStr": "[DEBUG] [agent] [2024-02-15T15:16:32.355Z]",
    "extra": [
      "Agent service starting"
    ]
  }
  2024-02-15T10:16:32-05:00 [INFO] 0 unhandled notification client/registerCapability:
  {
    "registrations": [
      {
        "id": "7739cf44-0f90-40b4-96e6-e6696a9df033",
        "method": "workspace/didChangeWorkspaceFolders",
        "registerOptions": {}
      }
    ]
  }
  2024-02-15T10:16:32-05:00 [INFO] 0 unhandled notification LogMessage:
  {
    "level": 0,
    "message": "[DEBUG] [agent] [2024-02-15T15:16:32.672Z] Telemetry initialized",
    "metadataStr": "[DEBUG] [agent] [2024-02-15T15:16:32.672Z]",
    "extra": [
      "Telemetry initialized"
    ]
  }
  2024-02-15T10:17:39-05:00 [ERROR] {
    "thread": "main",
    "payload": "assertion failed: self.window.next_frame.reused_views.insert(view_id)",
    "location_data": {
      "file": "crates/gpui/src/window/element_cx.rs",
      "line": 322
    },
    "backtrace": [
      "<backtrace::capture::Backtrace>::create",
      "<backtrace::capture::Backtrace>::new",
      "Zed::init_panic_hook::{closure#0}",
      "std::panicking::rust_panic_with_hook",
      "std::panicking::begin_panic_handler::{{closure}}",
      "std::sys_common::backtrace::__rust_end_short_backtrace",
      "_rust_begin_unwind",
      "core::panicking::panic_fmt",
      "core::panicking::panic",
      "<gpui::window::element_cx::ElementContext>::reuse_view",
      "<gpui::view::AnyView as gpui::element::Element>::paint",
      "<gpui::element::DrawableElement<gpui::view::AnyView>>::paint::{closure#0}",
      "<gpui::element::DrawableElement<gpui::view::AnyView>>::paint",
      "<core::option::Option<gpui::element::DrawableElement<gpui::view::AnyView>> as gpui::element::ElementObject>::paint",
      "<gpui::elements::div::Interactivity>::paint::<<gpui::elements::div::Div as gpui::element::Element>::paint::{closure#0}>::{closure#2}::{closure#0}::{closure#0}::{closure#0}",
      "<gpui::elements::div::Interactivity>::paint::<<gpui::elements::div::Div as gpui::element::Element>::paint::{closure#0}>::{closure#2}::{closure#0}::{closure#0}",
      "<gpui::elements::div::Div as gpui::element::Element>::paint",
      "<gpui::element::DrawableElement<gpui::elements::div::Div>>::paint",
      "<core::option::Option<gpui::element::DrawableElement<gpui::elements::div::Div>> as gpui::element::ElementObject>::draw",
      "<workspace::pane_group::element::PaneAxisElement as gpui::element::Element>::paint",
      "<gpui::element::DrawableElement<workspace::pane_group::element::PaneAxisElement>>::paint",
      "<core::option::Option<gpui::element::DrawableElement<workspace::pane_group::element::PaneAxisElement>> as gpui::element::ElementObject>::paint",
      "<gpui::element::DrawableElement<gpui::element::AnyElement>>::paint",
      "<core::option::Option<gpui::element::DrawableElement<gpui::element::AnyElement>> as gpui::element::ElementObject>::paint",
      "<gpui::element::DrawableElement<gpui::element::AnyElement>>::paint",
      "<core::option::Option<gpui::element::DrawableElement<gpui::element::AnyElement>> as gpui::element::ElementObject>::paint",
      "<gpui::elements::div::Interactivity>::paint::<<gpui::elements::div::Div as gpui::element::Element>::paint::{closure#0}>::{closure#2}::{closure#0}::{closure#0}::{closure#0}",
      "<gpui::elements::div::Interactivity>::paint::<<gpui::elements::div::Div as gpui::element::Element>::paint::{closure#0}>::{closure#2}::{closure#0}::{closure#0}",
      "<gpui::elements::div::Div as gpui::element::Element>::paint",
      "<gpui::element::DrawableElement<gpui::elements::div::Div>>::paint",
      "<core::option::Option<gpui::element::DrawableElement<gpui::elements::div::Div>> as gpui::element::ElementObject>::paint",
      "<gpui::elements::div::Interactivity>::paint::<<gpui::elements::div::Div as gpui::element::Element>::paint::{closure#0}>::{closure#2}::{closure#0}::{closure#0}::{closure#0}",
      "<gpui::elements::div::Interactivity>::paint::<<gpui::elements::div::Div as gpui::element::Element>::paint::{closure#0}>::{closure#2}::{closure#0}::{closure#0}",
      "<gpui::elements::div::Div as gpui::element::Element>::paint",
      "<gpui::element::DrawableElement<gpui::elements::div::Div>>::paint",
      "<core::option::Option<gpui::element::DrawableElement<gpui::elements::div::Div>> as gpui::element::ElementObject>::paint",
      "<gpui::elements::div::Interactivity>::paint::<<gpui::elements::div::Div as gpui::element::Element>::paint::{closure#0}>::{closure#2}::{closure#0}::{closure#0}::{closure#0}",
      "<gpui::elements::div::Interactivity>::paint::<<gpui::elements::div::Div as gpui::element::Element>::paint::{closure#0}>::{closure#2}::{closure#0}::{closure#0}",
      "<gpui::elements::div::Div as gpui::element::Element>::paint",
      "<gpui::element::DrawableElement<gpui::elements::div::Stateful<gpui::elements::div::Div>>>::paint::{closure#0}",
      "<gpui::window::element_cx::ElementContext>::with_element_state::<gpui::elements::div::DivState, (), <gpui::element::DrawableElement<gpui::elements::div::Stateful<gpui::elements::div::Div>>>::paint::{closure#0}>::{closure#0}",
      "<gpui::element::DrawableElement<gpui::elements::div::Stateful<gpui::elements::div::Div>>>::paint",
      "<core::option::Option<gpui::element::DrawableElement<gpui::elements::div::Stateful<gpui::elements::div::Div>>> as gpui::element::ElementObject>::paint",
      "<gpui::elements::div::Interactivity>::paint::<<gpui::elements::div::Div as gpui::element::Element>::paint::{closure#0}>::{closure#2}::{closure#0}::{closure#0}::{closure#0}",
      "<gpui::elements::div::Interactivity>::paint::<<gpui::elements::div::Div as gpui::element::Element>::paint::{closure#0}>::{closure#2}::{closure#0}::{closure#0}",
      "<gpui::elements::div::Div as gpui::element::Element>::paint",
      "<gpui::element::DrawableElement<gpui::elements::div::Div>>::paint",
      "<core::option::Option<gpui::element::DrawableElement<gpui::elements::div::Div>> as gpui::element::ElementObject>::paint",
      "<gpui::view::AnyView>::draw::{closure#0}",
      "<gpui::view::AnyView>::draw",
      "<gpui::window::WindowContext>::draw",
      "<gpui::window::Window>::new::{closure#0}::{closure#1}",
      "<gpui::window::Window>::new::{closure#0}",
      "gpui::platform::mac::window::step",
      "<gpui::platform::mac::platform::MacPlatform as gpui::platform::Platform>::run",
      "Zed::main",
      "std::sys_common::backtrace::__rust_begin_short_backtrace::<fn(), ()>",
      "std::rt::lang_start::<()>::{closure#0}",
      "std::rt::lang_start_internal",
      "_main"
    ],
    "app_version": "0.123.1",
    "release_channel": "Zed Preview",
    "os_name": "macOS",
    "os_version": "14.2.1",
    "architecture": "aarch64",
    "panicked_on": 1708010259777,
    "installation_id": "5a42fa36-7f10-4924-b350-cf41bd59ae2a",
    "session_id": "ecab9f5a-abee-4b2a-907c-d81bca6837b6"
  }
</details>

--- a/crates/extensions_ui/src/extensions_ui.rs
+++ b/crates/extensions_ui/src/extensions_ui.rs
@@ -1,12 +1,11 @@
 use client::telemetry::Telemetry;
 use editor::{Editor, EditorElement, EditorStyle};
 use extension::{Extension, ExtensionStatus, ExtensionStore};
-use fs::Fs;
 use gpui::{
     actions, canvas, uniform_list, AnyElement, AppContext, AvailableSpace, EventEmitter,
     FocusableView, FontStyle, FontWeight, InteractiveElement, KeyContext, ParentElement, Render,
-    Styled, Task, TextStyle, UniformListScrollHandle, View, ViewContext, VisualContext, WeakView,
-    WhiteSpace, WindowContext,
+    Styled, Task, TextStyle, UniformListScrollHandle, View, ViewContext, VisualContext, WhiteSpace,
+    WindowContext,
 };
 use settings::Settings;
 use std::time::Duration;
@@ -32,8 +31,6 @@ pub fn init(cx: &mut AppContext) {
 }
 
 pub struct ExtensionsPage {
-    workspace: WeakView<Workspace>,
-    fs: Arc<dyn Fs>,
     list: UniformListScrollHandle,
     telemetry: Arc<Telemetry>,
     is_fetching_extensions: bool,
@@ -46,7 +43,7 @@ pub struct ExtensionsPage {
 
 impl ExtensionsPage {
     pub fn new(workspace: &Workspace, cx: &mut ViewContext<Workspace>) -> View<Self> {
-        let extensions_panel = cx.new_view(|cx: &mut ViewContext<Self>| {
+        cx.new_view(|cx: &mut ViewContext<Self>| {
             let store = ExtensionStore::global(cx);
             let subscription = cx.observe(&store, |_, _, cx| cx.notify());
 
@@ -54,8 +51,6 @@ impl ExtensionsPage {
             cx.subscribe(&query_editor, Self::on_query_change).detach();
 
             let mut this = Self {
-                fs: workspace.project().read(cx).fs().clone(),
-                workspace: workspace.weak_handle(),
                 list: UniformListScrollHandle::new(),
                 telemetry: workspace.client().telemetry().clone(),
                 is_fetching_extensions: false,
@@ -67,8 +62,7 @@ impl ExtensionsPage {
             };
             this.fetch_extensions(None, cx);
             this
-        });
-        extensions_panel
+        })
     }
 
     fn install_extension(
@@ -459,25 +453,9 @@ impl Item for ExtensionsPage {
     fn clone_on_split(
         &self,
         _workspace_id: WorkspaceId,
-        cx: &mut ViewContext<Self>,
+        _: &mut ViewContext<Self>,
     ) -> Option<View<Self>> {
-        Some(cx.new_view(|cx| {
-            let store = ExtensionStore::global(cx);
-            let subscription = cx.observe(&store, |_, _, cx| cx.notify());
-
-            ExtensionsPage {
-                fs: self.fs.clone(),
-                workspace: self.workspace.clone(),
-                list: UniformListScrollHandle::new(),
-                telemetry: self.telemetry.clone(),
-                is_fetching_extensions: false,
-                extensions_entries: Default::default(),
-                query_editor: self.query_editor.clone(),
-                _subscription: subscription,
-                query_contains_error: false,
-                extension_fetch_task: None,
-            }
-        }))
+        None
     }
 
     fn to_item_events(event: &Self::Event, mut f: impl FnMut(workspace::item::ItemEvent)) {

